Initium-Odp/war/odp/javascript/script.js

```diff
@@ -619,19 +619,30 @@ function deletePlayerHouse(eventObject, pathId)
 	});
 }
 
-function storeBuyItemNew(eventObject, itemName, itemPrice, itemId, saleItemId, characterId)
+function storeBuyItemNew(eventObject, itemName, itemPrice, itemId, saleItemId, characterId, quantity)
 {
-	confirmPopup("Buy Item", "Are you SURE you want to buy this <a class='clue' rel='/viewitemmini.jsp?itemId="+itemId+"'>"+itemName+"</a> for "+itemPrice+" gold?", function(){
-		doCommand(eventObject, "StoreBuyItem",{"saleItemId":saleItemId,"characterId":characterId},function(data,error){
+	var yesFunction = function(qty=null){
+		doCommand(eventObject, "StoreBuyItem",{"saleItemId":saleItemId,"characterId":characterId,"quantity":qty},function(data,error){
 			if (error) return;
 			$(".saleItem[ref='"+saleItemId+"']").html(data.createStoreItem);
 		});
-	});
+	};
+	
+	if(quantity == 1)
+		confirmPopup("Buy Item", "Are you SURE you want to buy this <a class='clue' rel='/viewitemmini.jsp?itemId="+itemId+"'>"+itemName+"</a> for "+itemPrice+" gold?", yesFunction);
+	else{
+		rangePopup("Buy Item", "Please specify the number of <a class='clue' rel='/viewitemmini.jsp?itemId="+itemId+"'>"+itemName+"</a> to purchase ("+itemPrice+" gold each):",0,quantity,
+			function(qty) { // textFunction
+				return "Total cost: " + (qty*itemPrice);
+			},
+			yesFunction,
+			null); // noFunction
+	}
 }
 
 function storeSellItemNew(eventObject,itemId)
 {
-	promptPopup("Sell Item", "How much do you want to sell this item for?", "0", function(amount){
+	promptPopup("Sell Item", "How much do you want to sell this item for? (Unit Price)", "0", function(amount){
```

Comment:
I still find my version a bit unclear... if you have better wording let me know.